src/core/thread/mle.cpp

```diff
@@ -1396,7 +1396,7 @@ otError Mle::AppendCslChannel(Message &aMessage)
     otError       error = OT_ERROR_NONE;
     CslChannelTlv cslChannel;
 
-    VerifyOrExit(Get<Mac::Mac>().GetPanChannel() != Get<Mac::Mac>().GetCslChannel(), OT_NOOP);
+    VerifyOrExit(Get<Mac::Mac>().IsCslChannelSpecified(), OT_NOOP);
```

Comment:
What if the specified csl channel is the same as the pan channel?

Reply:
I think it doesn't matter. It's not forbidden in spec to specify the CSL channel to current pan channel. Though there's no effect in terms of CSL transmission, but the internal state(if CSL channel has been specified) should be changed in this scenario. Thoughts?

Reply:
> The Child Update Request SHALL include the CSL Channel TLV if the SSED selected a separate channel for CSL operation. Otherwise, the Child Update Request MAY include the CSL Channel TLV

agree that thread spec does not forbid the csl channel same as pan channel.